### PR TITLE
Add circular buffer exercise

### PR DESCRIPTION
**Pull request by kytrinyx**
This is a cherry-pick from #29 via @lehoff and @ebengt

### CODE REVIEW

**Comment by kytrinyx** [comment]
I'm installing erlang to see if I can get the tests running locally.

File: circular-buffer/circular_buffer_tests.erl [hunk 1]
```diff
@@ -0,0 +1,54 @@
+-module( circular_buffer_tests ).
```
Comment (etrepum) [inline circular-buffer/circular_buffer_tests.erl]: This is the problem, the test runner expects this to be `-module(circular_buffer_tests).`

File: circular-buffer/example.erl [hunk 2]
```diff
@@ -0,0 +1,86 @@
+-module( circular_buffer ).
```
Comment (etrepum) [inline circular-buffer/example.erl]: Same here, no spaces should be here. Doesn't matter for Erlang, but the test runner is fragile.

**Comment by etrepum** [comment]
I just improved the test runner a bit, the CI might pass next run. I'll try and trigger another run.

**Comment by kytrinyx** [comment]
It looks like there are compilation warnings, as well.

**Comment by kytrinyx** [comment]
I didn't squash the commits since different people wrote them.

**Comment by tmcgilchrist** [comment]
Nice :+1:

**Pull request merged**